### DISCUSSION TITLE
Strict CSP implementation

### DISCUSSION THREAD

**Discussion by padmakr** [vercel/next.js#22225] [Help]
**Describe the feature you'd like to request**

Currently we are using NEXT 10.0.6
We are unable to implement a CSP without "unsafe-inline" for the application. 

This is our current CSP
const CSP = `default-src 'self' https://www.googletagmanager.com/  'unsafe-eval' 'unsafe-inline' ;
script-src 'self' 'unsafe-eval' 'unsafe-inline' ;
script-src-elem 'self'  https://www.googletagmanager.com/  'unsafe-eval' 'unsafe-inline' ; 
img-src 'self'   www.googletagmanager.com;   
style-src 'self' * https://p.typekit.net/p.css https://use.typekit.net/ https://fonts.googleapis.com/css https://use.typekit.net/wyq6hey.css https://fonts.gstatic.com 'unsafe-inline';`

**Describe the solution you'd like**

NEXT js should support a CSP without having to use "unsafe-inline"

**Describe alternatives you've considered**

Unable to bypass this requirement of "unsafe-inline"

**Comment by josephsijoy** [comment]
We have been trying to add CSP headers to out next.js application to prevent click jacking attacks, since next.js uses inline css, adding the CSP headers breaks the application. Hosting an application which is vulnerable to attacks is not an option for us, please suggest a workaround or a permanent fix for this issue.

**Comment by FernandoBoza** [comment]
Is this in the pipeline?
